wait action fails

```
[INFO] [WallTime: 1427898715.389491] Action wait_action started for task 2
[INFO] [WallTime: 1427898715.410265] target wait time: 2015-04-01 15:32:25
[ERROR] [WallTime: 1427898715.411067] Exception in your execute callback: global name 'WaitFeedback' is not defined
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/dist-packages/actionlib/simple_action_server.py", line 299, in executeLoop
    self.execute_callback(goal)
  File "/home/strands/1f_ws/src/strands_executive/wait_action/scripts/wait_node.py", line 49, in execute
    feedback = WaitFeedback()
NameError: global name 'WaitFeedback' is not defined
```

```
 [INFO] [WallTime: 1427898760.400250] Action wait_action started for task 3
[ERROR] [WallTime: 1427898760.500989] Exception in your execute callback: service [/wait_action/end_wait] already registe$
ed
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/dist-packages/actionlib/simple_action_server.py", line 299, in executeLoop
    self.execute_callback(goal)
  File "/home/strands/1f_ws/src/strands_executive/wait_action/scripts/wait_node.py", line 30, in execute
    Empty, self.end_wait)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 712, in __init__
    get_service_manager().register(self.resolved_name, self)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/service.py", line 116, in register
    raise ServiceException(err)
ServiceException: service [/wait_action/end_wait] already registered
```

@marc-hanheide it looks like this was introduced by your edits to the wait action

--- a/wait_action/scripts/wait_node.py
+++ b/wait_action/scripts/wait_node.py
@@ -2,10 +2,12 @@
 
 import rospy
 from wait_action.msg import WaitAction
+from wait_action.msg import WaitFeedback
 from datetime import *
 from std_srvs.srv import Empty, EmptyResponse
 from strands_executive_msgs import task_utils
 from strands_executive_msgs.abstract_task_server import AbstractTaskServer
+
 
 class WaitServer(AbstractTaskServer):
     def __init__(self):
@@ -28,44 +30,42 @@ class WaitServer(AbstractTaskServer):
     def execute(self, goal):
         end_wait_srv = rospy.Service('/wait_action/end_wait',
                                      Empty, self.end_wait)
+        try:
+            now = rospy.get_rostime()
+            target = goal.wait_until
+            if target.secs == 0:
+                target = now + goal.wait_duration
 
-        now = rospy.get_rostime()
-        target = goal.wait_until
-        if target.secs == 0:
-            target = now + goal.wait_duration
+            if target == now:
+                # 72 hours
+                a_really_long_time = rospy.Duration(60 * 60 * 24 * 3)
+                target = now + a_really_long_time
+                # rospy.loginfo("waiting a really long time")
 
-        if target == now:
-            # 72 hours
-            a_really_long_time = rospy.Duration(60 * 60 * 24 * 3)
-            target = now + a_really_long_time
-            # rospy.loginfo("waiting a really long time")
+            rospy.loginfo("target wait time: %s"
+                          % datetime.fromtimestamp(target.secs))
 
+            # how often to provide feedback
+            feedback_secs = 5.0
+            feedback = WaitFeedback()
+            # how long to wait
+            wait_duration_secs = target.secs - now.secs
 
-        rospy.loginfo("target wait time: %s" \
-            % datetime.fromtimestamp(target.secs))	
+            # the rate to publish feedback at/check time at
+            r = rospy.Rate(1.0/feedback_secs)
 
-        # how often to provide feedback 
-        feedback_secs = 5.0
-        feedback = WaitFeedback()
-        # how long to wait
-        wait_duration_secs = target.secs - now.secs
+            count = 1.0
+            feedback_steps = wait_duration_secs/feedback_secs
+            while not rospy.is_shutdown() and not self.server.is_preempt_requested() and rospy.get_rostime() < target:
+                feedback.percent_complete = count/feedback_steps
+                self.server.publish_feedback(feedback)
+                count += 1
+                r.sleep()
 
-        # the rate to publish feedback at/check time at
-        r = rospy.Rate(1.0/feedback_secs)
-
-        count = 1.0
-        feedback_steps = wait_duration_secs/feedback_secs
-        while not rospy.is_shutdown() and not self.server.is_preempt_requested() and rospy.get_rostime() < target:
-            feedback.percent_complete = count/feedback_steps
-            self.server.publish_feedback(feedback)
-            count += 1
-            r.sleep()
-
-        rospy.loginfo("waited until: %s" % datetime.fromtimestamp(rospy.get_rostime().secs))
-
-
-
-        end_wait_srv.shutdown()
+            rospy.loginfo("waited until: %s"
+                          % datetime.fromtimestamp(rospy.get_rostime().secs))
+        finally:
+            end_wait_srv.shutdown()
 
         if self.server.is_preempt_requested():
             self.server.set_preempted()
@@ -76,7 +76,7 @@ class WaitServer(AbstractTaskServer):
 if __name__ == '__main__':
     rospy.init_node("wait_node")
 
-# wait for simulated time to kick in as rospy.get_rostime() is 0 until first clock message received
+    # wait for simulated time to kick in as rospy.get_rostime() is 0 until first clock message received
     while not rospy.is_shutdown() and rospy.get_rostime().secs == 0:
         pass
 

